nemo/collections/nlp/models/language_modeling/megatron/t5_model.py

```diff
@@ -73,22 +77,77 @@ def forward(self, hidden_states, word_embeddings_weight):
 class T5Model(MegatronModule):
     """T5 Language model."""
 
-    def __init__(self, num_tokentypes=0, parallel_output=True):
+    def __init__(
+        self,
+        vocab_size,
+        hidden_size,
+        max_position_embeddings,
+        num_layers,
+        num_attention_heads,
+        ffn_hidden_size,
+        apply_query_key_layer_scaling=True,
+        kv_channels=None,
+        num_tokentypes=0,
+        parallel_output=True,
+        pre_process=True,
+        post_process=True,
+        init_method_std=0.02,
+        fp16_lm_cross_entropy=False,
+        use_cpu_initialization=False,
+        hidden_dropout=0.1,
+        fused_fp16=False,
```

Comment:
Perhaps it is better to have a fused_precision argument instead of fused_fp16 and fused_bf16?
We cannot set both to true anyways, and we might want to support in the future fp32 (for example).

Reply:
Good point! Lets move this to the refactor.